### ISSUE TITLE
`import/consistent-type-specifier-style` for exports

### ISSUE THREAD

**Issue by musjj** [import-js/eslint-plugin-import#3044]
Like `import/consistent-type-specifier-style`, but for exports.

Example:

❌ Invalid with `["error", "prefer-top-level"]`

```ts
export {Foo, type Bar};
```

✅ Valid with `["error", "prefer-top-level"]`

```ts
export type {Bar};
export {Foo};
```

**Comment by ljharb** [comment]
That seems like a good option to add to `consistent-type-specifier-style`.
